docs/tests_in_tilt.md

```diff
@@ -0,0 +1,109 @@
+layout: docs
+---
+With the Tilt `test` primitive, you can run your tests from Tilt itself. Take advantage of Tilt's file-watching and responsiveness, and the power and customizability of Starlark, to supercharge your testing workflow.
+
+## How to Use It
+Define tests in your Tiltfile with the `test` primitive:
+```python
+test('cart-tests', 'go test ./pkg/cart', deps=['./pkg/cart'])
+test('py-tests', 'pytest app/', deps=['app/'])
+test('slow-integration-test', 'tests/integration.sh',
+     trigger_mode=TRIGGER_MODE_MANUAL, auto_init=False)
+```
+
+`test` supports all the same arguments as `local_resource` [see API reference](https://docs.tilt.dev/api.html#api.local_resource). The only notable difference today is that for tests, parallelism is _on_ by default (i.e. the default of `allow_parallel=True`).
+
+Your tests will appear in their own section of the overview grid, and alerts will be reflected on the test cards and in the status bar, so that you know at a glance when something is wrong.
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/overview-with-tests.png" alt="the Tilt UI overview page, with several tests">
+    <figcaption>Tests appear in their own section of the overview, and test status is summarized in the status bar at the top of the page</figcaption>
+</figure>
+
+Just like any other resource managed by Tilt, you can drill into a test and see logs and other details.
+
+In the log view, you can filter the sidebar by item type (if, say, you want to see only tests, or you want to filter tests out and focus on your app in the cluster). You can also toggle "Alerts On Top" sorting, to make it even more visible when something has gone wrong and needs your attention
+
+<figure>
+    <img src="/assets/img/tests-in-tilt/sidebar-alerts-on-top.png" alt="sidebar displaying only tests, with alerts sorted to the top">
+    <figcaption>Here's an example of sidebar filtering (displaying only tests). In "Alerts on Top" mode, any cards with alerts are sorted to the top of the sidebar for easy visibility</figcaption>
+</figure>
+
+### When Tests Run
+Just as with a `local_resource`, you can associate files with a test via the `deps=[...]` parameter. When any of these file dependencies change, Tilt will re-execute the test:
+
+```python
+# A test depending on an entire directory
+test('cart-tests', 'go test ./pkg/cart', deps=['./pkg/cart'])
+
+# A test depending on specific relevant files
+test('unit-test-remittance', 'pytest remittance_test.py',
+     deps=['app/remittance.py', 'app/remittance_test.py', 'app/payment_utils.py'])
+```
+(More on how to intelligently set your dependencies below.)
+
+Sometimes, you don't want your test(s) to execute automatically. Maybe you want to turn off automatic-ness for all but a few tests you're iterating on, or prevent an expensive integration test from running all the time. Use the "auto" trigger in the UI to toggle automatic execution on or off for a given test.
```

Comment:
> automatic-ness
um HALP what do @landism

Reply:
"Maybe you want to iterate on a few tests and disable the rest"?